contributors/design-proposals/networking-iam.md

```diff
@@ -0,0 +1,125 @@
+# Network strategy for IAM well-known IPs
+
+## Introduction
+
+AWS and GCE both define a well-known IP, 169.254.169.254, on which a HTTP
+server that provides system information can be obtained.  Typically this
+includes a script for booting a machine, data about the instance type and
+configuration, and also often authorization information for access to the
+full cloud API.
+
+The metadata service information is therefore either sensitive or inaccurate
+for many pods.  Sensitive because credentials are intended for the node, not
+the pod; and inaccurate because the metadata might report more memory than the
+pod has available.  As such it is often desirable to block access to the
+metadata server.  In addition it can be helpful to serve alternative
+information, such as alternative cloud access credentials, so that a pod can
+have a defined role for accessing cloud APIs.  By using the metadata service,
+existing code will work unaltered and transparently.
+
+This proposal documents a simple "agreed configuration" which will allow IAM
+proxies to operate consistently, and describes a few future enhancements for
+consideration once the value has been demonstrated.
+
+## Existing work
+
+[#42192](https://github.com/kubernetes/kubernetes/pull/42192) in Kubernetes 1.6
+on GCE created some firewall rules that by default prevent pods from reaching
+169.254.169.254.  It does not affect processes not running in a container, nor
+pods with `hostNetwork=true`.  It does not support proxying to a service,
+though it may support proxying to a local daemonset.
+
+```
+iptables -N KUBE-METADATA-SERVER
+iptables -A FORWARD -p tcp -d 169.254.169.254 --dport 80 -j KUBE-METADATA-SERVER
+iptables -A KUBE-METADATA-SERVER -j DROP
+```
+
+[kube2iam](https://github.com/jtblin/kube2iam) is widely used on AWS to provide
+pods with a particular IAM role.  It relies on a daemonset running on every
+node, and an iptables rule such as
+
+```
+iptables \
+  --append PREROUTING \
+  --protocol tcp \
+  --destination 169.254.169.254 \
+  --dport 80 \
+  --in-interface docker0 \
+  --jump DNAT \
+  --table nat \
+  --to-destination `curl 169.254.169.254/latest/meta-data/local-ipv4`:8181
+```
+
+kube2iam defines a pod annotation for controlling the IAM role granted:
+`iam.amazonaws.com/role: role-name`
+
+## Proposal
+
+This proposal is intended to allow for integration of a solution such as
+kube2iam into the kubernetes networking infrastructure.
+
+Specifically:
+
+* The rules in [#42192](https://github.com/kubernetes/kubernetes/pull/42192)
+  should be applied at the node level by the installation system, before
+  kube-proxy is running.  This acts as a "last line of defense" in case pods are
+  started before kube-proxy rules are fully in place.
+
+* We propose adding a "well known service" for IAM, as we have for DNS.  The
+  DNS service currently runs on `.10`, so we propose reserving `.9` for IAM.
```

Comment:
Ah, apparently it is merely a gentlemen's agreement.  The only thing stopping you grabbing .10 is that the installation tool likely got their first.